### DISCUSSION TITLE
[Feature request] decide stuff to include/exclude on installation/update with .lst file

### DISCUSSION THREAD

**Discussion by Gigas002** [prasanthrangan/hyprdots#1032] [Ideas]
Hey! Been looking at `Hyprdesk v0.3.0` PR, looks awesome atm!
Though I'm going to have a bad time merging it into my fork...
The new [manage_config](https://github.com/prasanthrangan/hyprdots/pull/1025/files#diff-aadaf313b9534d25b80361b3ebf94e1aefb10426f1757c72306545ce355775de) gave me an impression, that we will finally be able to exclude things we don't want to install, is this correct? IMO, in a perfect scenario, user forks's changes should mostly be the installation file tweaks to include/exclude stuff they don't want.

While I love the project very much and respect the contributors, personally, the main thing I dislike about master branch is that it tweaks some stuff outside of "theme manager" area of responsibility. Like the stuff that is outside of `$HOME` directory and requires to use `sudo`, etc. I don't see the *easy* way to exclude/change this behavior, rather than removing whole files and all the calls to these scripts, which requires lots of code reworks and code-scrolling. That includes:

- nvidia driver installation. I prefer to handle/have control on this matters myself. Besides, it's not directly related to theme management process itself
- sddm theme tweaking. Not in `$HOME`, so it requires `sudo` and *may* be breaking (e.g. the recent `sddm` and plasma update broke all the themes, that were created with old `QML` version) 
- grub tweaking. May be breaking in some cases (was actually breaking for me first time I've installed hyprdots, hopefully I had backups). Not a theme manager responsibility at all
- ability to exclude `code` related theming stuff. I was thinking about using `$EDITOR` variable instead of `code`, since I use `vscodium`, but that would be breaking for users, that use completely different editor, e.g. `kate`, so I guess it's not a good solution
- same goes for `kitty`. While `$TERM` variable can be used, terminal application configs differs quite much

That's about it for whining for today. Would be glad if it's implemented (in case these changes are approvable in terms of project philosophy) and would try to help as much as I can.

**Comment by kRHYME7** [comment]
> Hey! Been looking at Hyprdesk v0.3.0 PR, looks awesome atm!
Though I'm going to have a bad time merging it into my fork...
The new [manage_config](https://github.com/prasanthrangan/hyprdots/pull/1025/files#diff-aadaf313b9534d25b80361b3ebf94e1aefb10426f1757c72306545ce355775de) gave me an impression, that we will finally be able to exclude things we don't want to install, is this correct? IMO, in a perfect scenario, user forks's changes should mostly be the installation file tweaks to include/exclude stuff they don't want.

I am sorry for the confusion I accidentally pushed  my own setup. But the WIP version is this. [This includes theming all QT apps I guess. ](https://github.com/prasanthrangan/hyprdots/tree/wallbash-v3)

Also, the options for the config list is auto generated by ``` Hyprdots upgrade ``` and should be modified manually .

> nvidia driver installation. I prefer to handle/have control on this matters myself. Besides, it's not directly related to theme management process itself

Yeah, I agree to this too. Maybe just a prompt to ask if script should modify or not.. 

>  sddm theme tweaking. Not in $HOME, so it requires sudo and may be breaking (e.g. the recent sddm and plasma update broke all the themes, that were created with old QML version)
grub tweaking. May be breaking in some cases (was actually breaking for me first time I've installed hyprdots, hopefully I had backups). Not a theme manager responsibility at all

No comment,  But It is also good to have a prompt for this 


> ability to exclude code related theming stuff. I was thinking about using $EDITOR variable instead of code, since I use vscodium, but that would be breaking for users, that use completely different editor, e.g. kate, so I guess it's not a good solution
Same goes for kitty. While $TERM variable can be used, terminal application configs differs quite much


For simplicity, by default, the  ``` ./keybinding.conf ```  file will not be overwritten if users decides to restore from repository. So we users can freely modify this. Having this as default helps new users to use it as is. IMO, Advance users can easily modify these variables.  



I believe the maintainer makes the installation script straightforward for a fresh installation of arch.  This pretty much works almost all the time. But a prompt for user to choose is much better IMO.

**Reply by Gigas002** [reply]
Well, either it's a prompt or a file to comment out stuff, having the choice is always better than not, so I'm both hands for it.
Btw, does hyprdots-ctl handle the forked repos as source for updating/etc somehow?

**Reply by kRHYME7** [reply]
> Well, either it's a prompt or a file to comment out stuff, having the choice is always better than not, so I'm both hands for it.
Btw, does hyprdots-ctl handle the forked repos as source for updating/etc somehow?

Yep, you can definitely use a different repo also you can switch branches.  I am bad at naming the options, so feel free to contribute. 



![image](https://github.com/prasanthrangan/hyprdots/assets/53417443/e754a4de-ca7d-452f-a044-449b9f35d79e)

``` Hyprdots restore link [path/to/a/Clone/Directory ]``` This one makes the path as the Clone Directory 

``` Hyprdots branch ``` to change branch of the repo 

``` Hyprdots updat ``` pull  the upstream branch.  ( TO make this 100% success on pulling the updates it stash the changes first then pull then unstashed it again )


``` Hyprdots upgrade ```  run the installation again.  but with the custom files  declared at ``` ~/.config/hyprdots-ctl/manage_cfg.lst ```




``` Hyprdost-install ``` A hidden feature for now that installs Hyprdots 

![image](https://github.com/prasanthrangan/hyprdots/assets/53417443/5e1a215f-65e0-49dc-adb1-552a2c03613c)
